### PR TITLE
Add moth plush

### PR DESCRIPTION
**Pull request by Cenrus**
https://github.com/tgstation/tgstation/pull/49732, 46062, and 49878

## About this PR

Adds moth plushie and it's respective suicide act

:cl: ATHATH, ArcaneDefence
add: Moth plushie, found in arcade machines as a reward. Commiting suicide with it is ill advised.
/:cl:

### CODE REVIEW

**Comment by Bestparty** [comment]
umm no naked moth plushie thats erp......

**Comment by St0rmC4st3r** [comment]
What a great and intoxicating innocence?

**Comment by zeskorion** [comment]
Allow the banhammer to kill divine plushies

**Comment by Blueturbo47** [comment]
Time for felind plush, someones gonna make it the punching noise arent they

File: code/game/objects/items/plushes.dm [hunk 1]
```diff
@@ -118,9 +120,24 @@
...
+				if(iscarbon(user))
+					var/mob/living/carbon/C = user
+					if(C.drunkenness < 50)
+						C.drunkenness = min(C.drunkenness + 20, 50)
+				var/turf/current_location = get_turf(user)
+				var/area/current_area = current_location.loc //copied from hand tele code
+				if(current_location && current_area && current_area.noteleport)
+					to_chat(user, "<span class='notice'>There is no escape. No recall or intervention can work in this place.</span>")
+				else
+					to_chat(user, "<span class='notice'>There is no escape. Although recall or intervention can work in this place, attempting to flee from [src]'s immense power would be futile.</span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/plushes.dm]: This seems oddly specific to me, just sounds a bit weird with the whoole 'There is not escape. Although technically you could but you shouldn't' thing

**Review by PowerfulBacon — DISMISSED** [review]

File: code/game/objects/items/plushes.dm (hunk 1 above)
Comment (St0rmC4st3r) [reply code/game/objects/items/plushes.dm]: It's a pretty cool morrowind reference

File: code/game/objects/items/plushes.dm [hunk 2]
```diff
@@ -530,3 +549,30 @@
...
+	desc = "A plushie depicting an adorable mothperson. It's a huggable bug!"
+	icon_state = "moffplush"
+	item_state = "moffplush"
+	attack_verb = list("fluttered", "flapped")
+	squeak_override = list('sound/voice/moth/scream_moth.ogg'=1)
+///Used to track how many people killed themselves with item/toy/plush/moth
+	var/suicide_count = 0
+
+/obj/item/toy/plush/moth/suicide_act(mob/living/user)
+	user.visible_message("<span class='suicide'>[user] stares deeply into the eyes of [src] and it begins consuming [user.p_them()]!  It looks like [user.p_theyre()] trying to commit suicide!</span>")
```
Comment (Crossedfall) [inline code/game/objects/items/plushes.dm]: This is poorly worded. It sounds like the user is eating the plushy, when the plushy is actually consuming the user (how does that work?).

File: code/game/objects/items/plushes.dm [hunk 3]
```diff
@@ -530,3 +549,30 @@
...
+	suicide_count++
+	if(suicide_count < 3)
+		desc = "A plushie depicting an unsettling mothperson. After killing [suicide_count] [suicide_count == 1 ? "person" : "people"] it's not looking so huggable now..."
+	else
+		desc = "A plushie depicting a creepy mothperson. It's killed [suicide_count] people! I don't think I want to hug it any more!"
```
Comment (Crossedfall) [inline code/game/objects/items/plushes.dm]:
```suggestion
		desc = "A creepy mothperson plushy. It has killed [suicide_count] people! I don't think I want to hug it any more!"
```

File: code/game/objects/items/plushes.dm [hunk 4]
```diff
@@ -530,3 +549,30 @@
...
+	attack_verb = list("fluttered", "flapped")
+	squeak_override = list('sound/voice/moth/scream_moth.ogg'=1)
+///Used to track how many people killed themselves with item/toy/plush/moth
+	var/suicide_count = 0
+
+/obj/item/toy/plush/moth/suicide_act(mob/living/user)
+	user.visible_message("<span class='suicide'>[user] stares deeply into the eyes of [src] and it begins consuming [user.p_them()]!  It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	suicide_count++
+	if(suicide_count < 3)
+		desc = "A plushie depicting an unsettling mothperson. After killing [suicide_count] [suicide_count == 1 ? "person" : "people"] it's not looking so huggable now..."
```
Comment (Crossedfall) [inline code/game/objects/items/plushes.dm]:
```suggestion
		desc = "An unsettling mothperson plushy. After killing [suicide_count] [suicide_count == 1 ? "person" : "people"] it's not looking so huggable now..."
```

File: code/game/objects/items/plushes.dm [hunk 5]
```diff
@@ -530,3 +549,30 @@
 	attack_verb = list("stung")
 	gender = FEMALE
 	squeak_override = list('sound/voice/moth/scream_moth.ogg'=1)
+
+/obj/item/toy/plush/moth
+	name = "moth plushie"
+	desc = "A plushie depicting an adorable mothperson. It's a huggable bug!"
```
Comment (Crossedfall) [inline code/game/objects/items/plushes.dm]:
```suggestion
	desc = "An adorable mothperson plushy. It's a huggable bug!"
```

Weirdly phrased

**Review by Crossedfall — CHANGES_REQUESTED** [review]
Couple of phrasing/grammar issues that I noticed

**Review by qwertyquerty — CHANGES_REQUESTED** [review]
Remove the reference

**Comment by FargothUr** [comment]
The reference is funny stfu

**Review by PowerfulBacon — DISMISSED** [review]
The reference sounds bad and doesn't even show up online even after searching it

**Comment by Crossedfall** [comment]
> 
> 
> The reference sounds bad and doesn't even show up online even after searching it

It's Dagoth my dude. 

https://www.youtube.com/watch?v=WbaQ1nCD_dA
https://youtu.be/iR-K2rUP86M

**Comment by Cenrus** [comment]
reference was removed.

**Review by qwertyquerty — APPROVED** [review]

**Review by qwertyquerty — CHANGES_REQUESTED** [review]
Unused icon states?

**Comment by Cenrus** [comment]
forgot to remove them

**Review by Crossedfall — APPROVED** [review]

**Pull request merged**